pkgs/by-name/op/opensplat/package.nix

```diff
@@ -0,0 +1,84 @@
+{
+  lib,
+  gcc12Stdenv,
+  stdenv,
+  cmake,
+  ninja,
+  fetchFromGitHub,
+  fetchpatch,
+  python3,
+  opencv,
+  nlohmann_json,
+  nanoflann,
+  glm,
+  cxxopts,
+  config,
+  # Upstream has rocm/hip support, too. anyone?
+  cudaSupport ? config.cudaSupport,
+  cudaPackages,
+  autoAddDriverRunpath,
+  cmakeFlags ? [ ], # There's some interesting flags like OPENSPLAT_BUILD_SIMPLE_TRAINER or OPENSPLAT_MAX_CUDA_COMPATIBILITY
+}:
+let
+  version = "1.1.2";
+  torch = python3.pkgs.torch.override { inherit cudaSupport; };
+  # Using a normal stdenv with cuda torch gives
+  # ld: /nix/store/k1l7y96gv0nc685cg7i3g43i4icmddzk-python3.11-torch-2.2.1-lib/lib/libc10.so: undefined reference to `std::ios_base_library_init()@GLIBCXX_3.4.32'
+  mkDerivation = (if cudaSupport then gcc12Stdenv else stdenv).mkDerivation;
+in
+mkDerivation {
+  pname = "opensplat";
+  version = version;
+
+  src = fetchFromGitHub {
+    owner = "pierotofy";
+    repo = "OpenSplat";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-3tk62b5fSf6wzuc5TwkdfAKgUMrw3ZxetCJa2RVMS/s=";
+  };
+  patches = [
+    (fetchpatch {
+      name = "install-executables.patch";
+      url = "https://github.com/pierotofy/OpenSplat/commit/b4c4176819b508978583b7ebf66306171807a8e6.patch";
+      hash = "sha256-BUgPMcO3lt3ZEzv24u36k3aTEIoloOhxrCGi1KQ5Epk=";
+    })
+  ];
+  postPatch = ''
+    # the two vendored gsplats are so heavily modified they may be considered a fork
+    find vendor ! -name 'gsplat*' -maxdepth 1 -mindepth 1 -exec rm -rf {} +
+    mkdir vendor/{nanoflann,glm}
+    ln -s ${glm}/include/glm vendor/glm/glm
+    ln -s ${nanoflann}/include/nanoflann.hpp vendor/nanoflann/nanoflann.hpp
+    ln -s ${nlohmann_json}/include/nlohmann vendor/json
+    ln -s ${cxxopts}/include/cxxopts.hpp vendor/cxxopts.hpp
+  '';
+
+  nativeBuildInputs =
+    [ cmake ninja ]
```

Comment:
```suggestion
  nativeBuildInputs =[ 
    cmake 
    ninja
 ]
```

Reply:
Are you using some specific formatter for this? (Probably not, given the whitespacing error there. I think I used nixfmt-rfc-style. Having the two lists at different indentation levels looks weird to me, but oh well. I applied the style you suggest anyway.)

Reply:
I was using the cellphone during rush, sorry the horrible.

Well, the idea is that lists should be listed one element per line. It helps making diffs smaller, since a new element will generate just a green line and a removal will generate a red/pink line.

Usually nixfmt is conservative about breaking lines.

Reply:
Ah. Coding on a cellphone isn't fun. I appreciate getting a review ^^.